src/cpp/r/RCntxt.cpp

```diff
@@ -81,6 +81,41 @@ bool RCntxt::hasSourceRefs() const
    return refs && TYPEOF(refs) != NILSXP;
 }
 
+SEXP RCntxt::contextSourceRefs() const
+{
+   // retrieve the source reference tagged on the context
+   SEXP ref = srcref();
+   
+   // if this is a byte-code context, we need to do some
+   // extra work to resolve the "real" source reference (if any)
+   if (isByteCodeSrcRef(ref))
+   {
+      r::sexp::Protect protect;
+   
+      // stuff context parameters into a list to avoid accidentally
+      // evaluating call objects prematurely
+      r::sexp::ListBuilder builder(&protect);
+      builder.add("call", call());
+      builder.add("callflag", callflag());
+      builder.add("callfun", callfun());
+      builder.add("srcref", srcref());
+      builder.add("cloenv", cloenv());
+      SEXP info = r::sexp::create(builder, &protect);
```

Comment:
It doesn't look like the the function actually uses most of these values -- do we need them all?

Reply:
We don't; I just took the kitchen sink because I wasn't sure what I'd need. I've narrowed this down so we only pass along the essentials.